public/pages/Rollups/containers/Rollups/Rollups.tsx

```diff
@@ -0,0 +1,583 @@
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import React, { Component } from "react";
+import _ from "lodash";
+import chrome from "ui/chrome";
+import { BREADCRUMBS, ROUTES } from "../../../../utils/constants";
+import queryString from "query-string";
+import { toastNotifications } from "ui/notify";
+import { getErrorMessage } from "../../../../utils/helpers";
+import { ManagedCatIndex } from "../../../../../server/models/interfaces";
+import { DEFAULT_PAGE_SIZE_OPTIONS, DEFAULT_QUERY_PARAMS } from "../../../Indices/utils/constants";
+import { RouteComponentProps } from "react-router-dom";
+import {
+  EuiBasicTable,
+  EuiHorizontalRule,
+  // @ts-ignore
+  Criteria,
+  EuiTableSortingType,
+  Direction,
+  // @ts-ignore
+  Pagination,
+  EuiTableSelectionType,
+  EuiFlexItem,
+  EuiFieldSearch,
+  EuiPagination,
+  EuiFlexGroup,
+  EuiPanel,
+  EuiTitle,
+  EuiButton,
+  EuiPopover,
+  EuiContextMenuItem,
+  EuiContextMenuPanel,
+  EuiTextColor,
+  EuiLink,
+  EuiTableFieldDataColumnType,
+  EuiIcon,
+  EuiText,
+} from "@elastic/eui";
+import { RollupService } from "../../../../services";
+import RollupEmptyPrompt from "../../components/RollupEmptyPrompt";
+import { RollupItem, RollupsQueryParams } from "../../models/interfaces";
+import { getURLQueryParams, renderContinuous, renderEnabled, renderTime } from "../../utils/helpers";
+import DeleteModal from "../../components/DeleteModal";
+
+interface RollupsProps extends RouteComponentProps {
+  rollupService: RollupService;
+}
+
+interface RollupsState {
+  totalRollups: number;
+  from: number;
+  size: number;
+  search: string;
+  sortField: keyof RollupItem;
+  sortDirection: Direction;
+  selectedItems: RollupItem[];
+  rollups: RollupItem[];
+  rollupExplain: any;
+  loadingRollups: boolean;
+  isPopoverOpen: boolean;
+  isDeleteModalVisible: boolean;
+}
+
+export default class Rollups extends Component<RollupsProps, RollupsState> {
+  constructor(props: RollupsProps) {
+    super(props);
+
+    const { from, size, search, sortField, sortDirection } = getURLQueryParams(this.props.location);
+
+    this.state = {
+      totalRollups: 0,
+      from,
+      size,
+      search,
+      sortField,
+      sortDirection,
+      selectedItems: [],
+      rollups: [],
+      loadingRollups: false,
+      isPopoverOpen: false,
+      isDeleteModalVisible: false,
+      rollupExplain: {},
+    };
+
+    this.getRollups = _.debounce(this.getRollups, 500, { leading: true });
+  }
+
+  async componentDidMount() {
+    chrome.breadcrumbs.set([BREADCRUMBS.INDEX_MANAGEMENT, BREADCRUMBS.ROLLUPS]);
+    await this.getRollups();
+    const { rollups } = this.state;
+    if (rollups.length != 0) await this.getExplains();
```

Comment:
Can we move the combined getRollups and getExplains merging to a single backend API?